examples/Rules-method-nextBest-NextBestProbMTDLTE.R

```diff
@@ -0,0 +1,41 @@
+# Example of usage for `NextBestProbMTDLTE` NextBest class.
+
+# Create the data.
+my_data <- Data(
+  x = c(0.1, 0.5, 1.5, 3, 6, 10, 10, 10),
+  y = c(0, 0, 0, 0, 0, 0, 1, 0),
+  ID = 1:8,
+  cohort = c(0, 1, 2, 3, 4, 5, 5, 5),
+  doseGrid = c(0.1, 0.5, 1.5, 3, 6, seq(from = 10, to = 80, by = 2))
+)
+
+# Initialize the CRM model used to model the data.
+my_model <- LogisticLogNormal(
+  mean = c(-0.85, 1),
+  cov = matrix(c(1, -0.5, -0.5, 1), nrow = 2),
+  ref_dose = 56
+)
+
+# Set-up some MCMC parameters and generate samples from the posterior.
+my_options <- McmcOptions(burnin = 100, step = 2, samples = 500)
+my_samples <- mcmc(my_data, my_model, my_options)
+
+# Define the rule for dose increments and calculate the maximum dose allowed.
+my_increments <- IncrementsRelative(
+  intervals = c(0, 20),
+  increments = c(1, 0.33)
+)
+next_max_dose <- maxDose(my_increments, data = my_data)
+
+# Define the rule which will be used to select the next best dose
+# based on the 'NextBestProbMTDLTE' class.
+prob_mtd_next_best <- NextBestProbMTDLTE(target = 0.33)
+
+# Calculate the next best dose.
+dose_recommendation <- nextBest(
+  nextBest = prob_mtd_next_best,
+  doselimit = next_max_dose,
+  samples = my_samples,
+  model = my_model,
+  data = my_data
+)
```

Comment:
# Create the data with placebo
PL <- 0.001
data <- Data(x        = c(PL, 25, 25, 25, PL, 50, 50, 50, PL, 100, 100, 100),
             y        = c( 0,  0,  0,  0,  0,  0,  0,  0,  0,   0,    1,  0),
             cohort   = c( 1,  1,  1,  1,  2,  2,  2,  2,  3,   3,    3,  3),
             doseGrid = c(PL, seq(25, 300, 25)),
             ID       = 1:12,
             placebo  = TRUE
             )

# Initialize the CRM model used to model the data
model <- LogisticLogNormal(mean     = c(-0.85, 1),
                           cov      = matrix(c(1, -0.5, -0.5, 1),
                                            nrow=2
                                            ),
                           ref_dose = 100
                           )

# coarseGrid <- c(25, 50, 100, 200, 300)
# model <- MinimalInformative(dosegrid = coarseGrid,
#                             refDose  = 100,
#                             logNormal = TRUE,
#                             threshmin = 0.1,
#                             threshmax = 0.2,
#                             seed = 432,
#                             control = list(max.time = 30)
#                             )

# Set-up some MCMC parameters and generate samples from the posterior
options <- McmcOptions(burnin  = 1000,
                       step  = 2,
                       samples = 10000
                      )
set.seed(94)
samples <- mcmc(data, model, options)


# Define the rule for dose increments and calculate the maximum dose allowed
myIncrements <- IncrementsRelative(intervals  = c(0, 100, 200),
                                   increments = c(1, 0.5, 0.33)
)

nextMaxDose <- maxDose(myIncrements,
                       data = data
)

doselimit <- nextMaxDose

# Define the rule which will be used to select the next best dose
# based on the class 'NextBestProbMTDLTE'
mtdNextBest <- NextBestProbMTDLTE(target = 0.3,
                                  method = "max"
)

# Calculate the next best dose
doseRecommendation <- nextBest(mtdNextBest,
                               doselimit = nextMaxDose,
                               samples = samples,
                               model = model,
                               data = data
)